mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -209,14 +209,29 @@ public Receive createReceive() {
 
     private void onAddNewJobArtifactsToCacheRequest(AddNewJobArtifactsToCacheRequest req) {
         try {
-            mantisJobStore.addNewJobArtifactsToCache(req.getClusterID(), req.getArtifacts());
-            jobArtifactsToCache.addAll(req.artifacts);
+            List<ArtifactID> newArtifacts = req.artifacts
+                .stream()
+                .filter(artifactID -> !jobArtifactsToCache.contains(artifactID))
+                .collect(Collectors.toList());
+
+            if (!newArtifacts.isEmpty()) {
+                log.info("Storing and caching new artifacts: {}", newArtifacts);
+
+                mantisJobStore.addNewJobArtifactsToCache(req.getClusterID(), newArtifacts);
+                jobArtifactsToCache.addAll(newArtifacts);
```

Comment:
shall we track/alert the size of cached artifacts in e.g. metrics?

Reply:
done.